epytope/TCRSpecificityPrediction/External.py

```diff
@@ -408,3 +408,82 @@ def format_results(self, filenames, tcrs, epitopes, pairwise):
         joining_list = ["VDJ_cdr3", "Epitope"]
         df_out = self.transform_output(results_predictor, tcrs, epitopes, pairwise, joining_list)
         return df_out
+
+class Ergo1(ARepoTCRSpecificityPrediction):
+    """
+    Author: Springer et al.
+    Paper: https://www.frontiersin.org/articles/10.3389/fimmu.2020.01803/full
+    Repo: https://github.com/louzounlab/ERGO
+    """
+    __name = "ERGO-I"
+    __version = ""
+    __tcr_length = (0, 30)  # TODO found no info in paper
+    __epitope_length = (0, 30)  # TODO found no info in paper
+    __repo = "https://github.com/louzounlab/ERGO.git"
+
+    _rename_columns = {
+        "VDJ_cdr3": "CDR3b"
+    }
+
+    @property
+    def name(self):
+        return self.__name
+
+    @property
+    def version(self):
+        return self.__version
+
+    @property
+    def tcr_length(self):
+        return self.__tcr_length
+
+    @property
+    def epitope_length(self):
+        return self.__epitope_length
+
+    @property
+    def repo(self):
+        return self.__repo
+
+    def format_tcr_data(self, tcrs, epitopes, pairwise):
+        required_columns = list(self._rename_columns.values()) + ["epitope"]
+        df_tcrs = tcrs.to_pandas(rename_columns=self._rename_columns)
+        if pairwise:
+            df_tcrs = self.combine_tcrs_epitopes_pairwise(df_tcrs, epitopes)
+        else:
+            df_tcrs = self.combine_tcrs_epitopes_list(df_tcrs, epitopes)
+        df_tcrs = df_tcrs.rename(columns={"Epitope": "epitope"})
+        df_tcrs = df_tcrs[required_columns]
+        df_tcrs = self.filter_by_length(df_tcrs, None, "CDR3b", "epitope")
+        df_tcrs = df_tcrs.drop_duplicates()
+        df_tcrs = df_tcrs[(~df_tcrs["CDR3b"].isna()) & (df_tcrs["CDR3b"] != "")]
+        return df_tcrs
+    
+    def save_tmp_files(self, data, **kwargs):
+        tmp_folder = self.get_tmp_folder_path()
+        path_in = os.path.join(tmp_folder.name, f"{self.name}_input.csv")
+        path_out = os.path.join(tmp_folder.name, f"{self.name}_output.csv")
+        data.to_csv(path_in, index=False, header=False)
+        return [path_in, path_out], tmp_folder
+
+    def get_base_cmd(self, filenames, tmp_folder, interpreter=None, conda=None, cmd_prefix=None, **kwargs):
+        dataset = "vdjdb" if "dataset" not in kwargs else kwargs["dataset"]
+        model_type = "lstm" if "model_type" not in kwargs else kwargs["model_type"]
+        cuda = "cpu" if "cuda" not in kwargs else kwargs["cuda"]
+        sampling = "specific" if "sampling" not in kwargs else kwargs["sampling"] #do we need sampling parameter?
+        repository = kwargs["repository"]
+        model = "lstm_vdjdb1" #model_filepath auto does not work
```

Comment:
is model not determined by model_type and dataset? If so please remove redundancy. Otherwise model follows different code structure than eg. sampling with the multiline if statement. Please make it uniform (I prefer the oneliner).

Reply:
There is an option to set model to "auto". In this case sampling is use to determine the model name/path used for prediction. The problem is that this "auto" setting does not work or works only in some cases, because they didn't publish models with respective names. So, I remove it?

Reply:
Yes. What happens if the user specifies dataset and model_type and a different model? I think one of that should be redundant, no?